app/messages/validators.rb

```diff
@@ -158,6 +158,14 @@ def validate(record)
     end
   end
 
+  class ReadinessHealthCheckValidator < ActiveModel::Validator
+    def validate(record)
+      if record.readiness_health_check_type != VCAP::CloudController::HealthCheckTypes::HTTP
+        record.errors.add(:readiness_health_check_type, message: 'must be "http" to set a health check HTTP endpoint')
```

Comment:
strangely, I can't get the equivalent validation for HealthCheckValidator to trigger:

```
± sb |readiness-health-checks-story-2-in-db → origin {3} ?:1 ✗| → cf curl -X PATCH v3/processes/c6493e8f-84b0-4e84-be1f-7156b7704039 -d '{"health_check": {"port": "process","data":  { "endpoint": "/bar" }}}'
{
   "guid": "c6493e8f-84b0-4e84-be1f-7156b7704039",
   "created_at": "2023-07-26T21:11:14Z",
   "updated_at": "2023-07-26T21:56:11Z",
   "type": "web",
   "command": "",
   "instances": 1,
   "memory_in_mb": 256,
   "disk_in_mb": 1024,
   "log_rate_limit_in_bytes_per_second": -1,
   "health_check": {
      "type": "process",
      "data": {
         "timeout": null,
         "invocation_timeout": null
      }
   },
   "relationships": {
      "app": {
         "data": {
            "guid": "c6493e8f-84b0-4e84-be1f-7156b7704039"
         }
      },
      "revision": null
   },
   "metadata": {
      "labels": {},
      "annotations": {}
   },
   "links": {
      "self": {
         "href": "https://api.vivid-carver.capi.land/v3/processes/c6493e8f-84b0-4e84-be1f-7156b7704039"
      },
      "scale": {
         "href": "https://api.vivid-carver.capi.land/v3/processes/c6493e8f-84b0-4e84-be1f-7156b7704039/actions/scale",
         "method": "POST"
      },
      "app": {
         "href": "https://api.vivid-carver.capi.land/v3/apps/c6493e8f-84b0-4e84-be1f-7156b7704039"
      },
      "space": {
         "href": "https://api.vivid-carver.capi.land/v3/spaces/3072c013-78fd-4c31-b9e8-1c80ad0f3a48"
      },
      "stats": {
         "href": "https://api.vivid-carver.capi.land/v3/processes/c6493e8f-84b0-4e84-be1f-7156b7704039/stats"
      }
   }
}
```

Guess we just run that validator on the manifest for some reason